### PR TITLE
Remove WorldUtils.AreMutualAllies

### PR DESCRIPTION
**Pull request by abcdefg30**
As suggested at https://github.com/OpenRA/OpenRA/pull/18677#discussion_r522408113.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/ProximityCapturable.cs [hunk 1]
```diff
@@ -153,7 +147,7 @@ void UpdateOwnership()
 			{
 				if (Info.MustBeClear)
 				{
-					var isClear = IsClear(Self, captor.Owner);
+					var isClear = actorsInRange.All(a => a.Owner == captor.Owner || a.Owner.RelationshipWith(captor.Owner) == PlayerRelationship.Ally);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ProximityCapturable.cs]: We can drop the `a.Owner == captor.Owner` check here now, right?

Comment (pchote) [reply OpenRA.Mods.Common/Traits/ProximityCapturable.cs]: and also use `.IsAlliedWith` to simplify even further (unless we are planning to remove that next too?)

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/ProximityCapturable.cs]: Updated. I think we should remove or replace `IsAlliedWith` so I left it as is for now.

File: OpenRA.Mods.Common/Traits/ProximityCapturable.cs [hunk 2]
```diff
@@ -153,7 +147,7 @@ void UpdateOwnership()
 			{
 				if (Info.MustBeClear)
 				{
-					var isClear = IsClear(Self, captor.Owner);
+					var isClear = actorsInRange.All(a => a.Owner.RelationshipWith(captor.Owner) == PlayerRelationship.Ally);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ProximityCapturable.cs]: Shouldn't this be `captor.Owner.RelationshipWith(a.Owner)`?

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/ProximityCapturable.cs]: Derp, yes.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/ProximityCapturable.cs]: Fixed again.

**Review by pchote — APPROVED** [review]

**Comment by deleted-user-1** [comment]
Forgot changelog?

**Comment by Mailaender** [comment]
No visible change for end users.

**Pull request merged**